src/main/kotlin/io/curity/identityserver/plugin/frejaeid/authentication/RequestLogicHelper.kt

```diff
@@ -0,0 +1,134 @@
+import se.curity.identityserver.sdk.service.WebServiceClient
+import java.net.URI
+import java.net.URLEncoder
+import java.nio.charset.StandardCharsets
+import java.util.*
+import kotlin.collections.HashMap
+import kotlin.collections.set
+
+class RequestLogicHelper(private val config: FrejaEidAuthenticatorPluginConfig)
+{
+    private val _json = config.json
+    private val _httpClient = config.httpClient
+    private val _exceptionFactory = config.exceptionFactory
+    private val _relyingPartyId = config.relyingPartyId.orElse(null)
+    private val _minRegistrationLevel = config.minimumRegistrationLevel
+    private val _attributesToReturn = config.attributesToReturn
+    private val _logger: Logger = LoggerFactory.getLogger(RequestLogicHelper::class.java)
+
+    companion object
+    {
+        const val QR_CODE_GENERATE_URL_TEST = "https://resources.test.frejaeid.com"
+        const val QR_CODE_GENERATE_URL_PROD = "https://resources.prod.frejaeid.com"
+        const val QR_CODE_GENERATE_PATH = "/qrcode/generate?qrcodedata="
+    }
+
+    fun createPostData(userInfoType: UserInfoType, username: String): Map<String, Any>
+    {
+        val dataMap = HashMap<String, Any>(4)
+
+        dataMap["userInfoType"] = userInfoType.toString()
+        dataMap["minRegistrationLevel"] = _minRegistrationLevel.toString()
+
+        if (userInfoType.equals(UserInfoType.SSN))
+        {
+            val userInfo = HashMap<String, Any>(2)
+            userInfo["country"] = "SE"
+            userInfo["ssn"] = username
+            dataMap["userInfo"] = Base64.getEncoder().encodeToString(_json.toJson(userInfo).toByteArray())
+        }
+        else
+        {
+            dataMap["userInfo"] = username
+        }
+
+        if (_minRegistrationLevel == RegistrationLevel.BASIC)
+        {
+            dataMap["attributesToReturn"] = _attributesToReturn.filter { attr ->
+                AttributesToReturn.EMAIL_ADDRESS == attr || AttributesToReturn.INTEGRATOR_SPECIFIC_USER_ID == attr
+            }.map { mapOf("attribute" to it) }
+        }
+        else if (_attributesToReturn.isNotEmpty() && _minRegistrationLevel != RegistrationLevel.BASIC)
+        {
+            dataMap["attributesToReturn"] = _attributesToReturn.map { mapOf("attribute" to it) }
+        }
+
+        return dataMap
+    }
+
+    fun getAuthTransaction(postData: Map<String, Any>): Map<String, Any>
+    {
+        val initialRequest = "initAuthRequest=${Base64.getEncoder().encodeToString(buildJsonAuthRequest(postData).toByteArray())}"
+        val requestBody = _relyingPartyId?.let { "$initialRequest&relyingPartyId=$it" } ?: initialRequest
+        val httpResponse = getWebServiceClient(config.environment.getHost())
+                .withPath("/authentication/1.0/initAuthentication")
+                .request()
+                .contentType("text/plain")
+                .body(HttpRequest.fromString(requestBody, StandardCharsets.UTF_8))
+                .method("POST")
```

Comment:
can change to `.post()` IINM. Nitpick sorry. Not an important change.

Reply:
Fixing in all places we do this